src/main/java/net/snowflake/ingest/streaming/internal/StageMetadata.java

```diff
@@ -0,0 +1,127 @@
+package net.snowflake.ingest.streaming.internal;
+
+import java.util.Map;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.annotation.JsonProperty;
+
+class StageMetadata {
+  @JsonProperty("locationType")
+  private String stageType; // The stage type
+
+  @JsonProperty("location")
+  private String location; // The container or bucket
+
+  @JsonProperty("path")
+  private String path; // path of the target file
+
+  @JsonProperty("creds")
+  private Map<?, ?> credentials; // the credentials required for the  stage
```

Comment:
curious, why is this `Map<?,?>` ? Do we not have an explicit type for the credentials response?

Reply:
Referenced from [here](https://github.com/snowflakedb/snowflake-jdbc/blob/6a95dccda713969cc8681f2c3bfa5791b4d8b97e/src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java#L1322). Just saw they are using String -> String mapper, changed to Map<String, String>.